controllers/model/model_controller.go

```diff
@@ -207,6 +210,13 @@ func (r *ModelReconciler) initializeContext(ctx *reconcileRequestContext) error
 	ctx.SageMakerClient = clientwrapper.NewSageMakerClientWrapper(r.createSageMakerClient(awsConfig))
 	ctx.Log.Info("Loaded AWS config")
 
+	if ctx.Model.Spec.PrimaryContainer.Mode == nil {
+		ctx.Model.Spec.PrimaryContainer.Mode = aws.String(DefaultContainerDefinitionMode)
+	}
+	for _, container := range ctx.Model.Spec.Containers {
+		container.Mode = aws.String(DefaultContainerDefinitionMode)
```

Comment:
@akartsky check if its nil before setting value ?

what is special in aws.String?

Reply:
oh wow 
nice catch

Reply:
oh wow.. nice catch 

-----------------------

aws.String is a helper function to assign strings 
https://docs.aws.amazon.com/sdk-for-go/api/aws/

**If you don't use it we get error message**

`container.Mode = &DefaultContainerDefinitionMode` : cannot take address of DefaultContainerDefinitionMode (untyped string constant "SingleModel")
`container.Mode = DefaultContainerDefinitionMode` : cannot convert DefaultContainerDefinitionMode (untyped string constant "SingleModel") to *string

Reply:
for sake of consistency can we use this - https://github.com/aws/amazon-sagemaker-operator-for-k8s/blob/master/controllers/controllertest/util.go#L32-L34